commit dabea74d0e82ea80cd344f630497cafcb3ef872c
Author: Michael Niedermayer <michael@niedermayer.cc>
Date:   Wed Sep 30 13:10:48 2015 +0200

    avcodec/vp8: Do not use num_coeff_partitions in thread/buffer setup
    
    The variable is not a constant and can lead to race conditions
    
    Fixes: repro.webm (not reproducable with FFmpeg alone)
    
    Found-by: Dale Curtis <dalecurtis@google.com>
    Tested-by: Dale Curtis <dalecurtis@google.com>
    Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>

diff --git a/libavcodec/vp8.c b/libavcodec/vp8.c
index 730871a12d..cb0c7cdc02 100644
--- a/libavcodec/vp8.c
+++ b/libavcodec/vp8.c
@@ -164,7 +164,7 @@ int update_dimensions(VP8Context *s, int width, int height, int is_vp7)
     s->mb_height = (s->avctx->coded_height + 15) / 16;
 
     s->mb_layout = is_vp7 || avctx->active_thread_type == FF_THREAD_SLICE &&
-                   FFMIN(s->num_coeff_partitions, avctx->thread_count) > 1;
+                   avctx->thread_count > 1;
     if (!s->mb_layout) { // Frame threading and one thread
         s->macroblocks_base       = av_mallocz((s->mb_width + s->mb_height * 2 + 1) *
                                                sizeof(*s->macroblocks));
